ps_linklist.php

```diff
@@ -255,28 +249,36 @@ private function addModuleErrors(array $errors)
     }
 
     /**
-     * @return LinkBlockRepository|null
+     * @return LinkBlockRepository|LegacyLinkBlockRepository|null
      */
     private function getRepository()
     {
-        if (null === $this->repository && $this->isSymfonyContext()) {
+        if (null === $this->repository) {
             try {
                 $this->repository = $this->get('prestashop.module.link_block.repository');
             } catch (\Exception $e) {
-                //Module is not installed so its services are not loaded
-                /** @var LegacyContext $context */
-                $legacyContext = $this->get('prestashop.adapter.legacy.context');
-                /** @var Context $shopContext */
-                $shopContext = $this->get('prestashop.adapter.shop.context');
-                $this->repository = new LinkBlockRepository(
-                    $this->get('doctrine.dbal.default_connection'),
-                    SymfonyContainer::getInstance()->getParameter('database_prefix'),
-                    $legacyContext->getLanguages(true, $shopContext->getContextShopID()),
-                    $this->get('translator')
-                );
+                try {
+                    //Module is not installed so its services are not loaded
+                    /** @var LegacyContext $context */
+                    $legacyContext = $this->get('prestashop.adapter.legacy.context');
+                    /** @var Context $shopContext */
+                    $shopContext = $this->get('prestashop.adapter.shop.context');
+                    $this->repository = new LinkBlockRepository(
+                        $this->get('doctrine.dbal.default_connection'),
+                        SymfonyContainer::getInstance()->getParameter('database_prefix'),
+                        $legacyContext->getLanguages(true, $shopContext->getContextShopID()),
+                        $this->get('translator')
+                    );
+                } catch (\Exception $e) {
```

Comment:
isn't a mandatory part to retrieve the repository?
This kind of `try.. catch` must be around the `getRepository` call, no?

Reply:
The idea is that `getRepository` will always return something, even if the container or service was not found it fallbacks by returning the `legacyBlockRepository` Which is why all `try...catch` block are in this method

This allows to use it more simply in install/uninstall functions by doing `$this->getRepository()->createTables()` for example (both repositories implement this method)

Reply:
Didn't notice the last part about `legacyBlockRepository` my bad :sweat_smile: